Sourcery/Parsing/Parser.swift

```diff
@@ -252,102 +259,79 @@ final class Parser {
         }
     }
 
-    private func processContainedType(_ type: Any, within containingType: Any) {
-        ///! only Type can contain children
-        guard let containingType = containingType as? Type else {
+    private func processContainedType(_ type: Any, within containing: Any) {
+        if let containingType = containing as? Type {
+            switch (containingType, type) {
+            case let (_, variable as Variable):
+                containingType.variables += [variable]
+                if !variable.isStatic {
+                    if let enumeration = containingType as? Enum,
+                        let updatedRawType = parseEnumRawType(enumeration: enumeration, from: variable) {
+                        
+                        enumeration.rawType = updatedRawType
+                    }
+                }
+                
+            case let (_, method as Method):
+                if method.isInitializer {
+                    method.returnTypeName = containingType.name
+                }
+                containingType.methods += [method]
+            case let (_, childType as Type):
+                containingType.containedTypes += [childType]
+                childType.parent = containingType
+            case let (enumeration as Enum, enumCase as Enum.Case):
+                enumeration.cases += [enumCase]
+            default:
+                break
+            }
+            
             return
         }
-
-        switch (containingType, type) {
-        case let (_, variable as Variable):
-            containingType.variables += [variable]
-            if !variable.isStatic {
-                if let enumeration = containingType as? Enum,
-                    let updatedRawType = parseEnumRawType(enumeration: enumeration, from: variable) {
-
-                    enumeration.rawType = updatedRawType
-                }
+        
+        if let containingMethod = containing as? Method {
```

Comment:
I'd reverse the order, put this on top of function since it's shorter and then the structure is a little easier to read